### ISSUE TITLE
Task with output streams

### ISSUE THREAD

**Issue by choubisat** [Netflix/conductor#1057]
Conductor allows executing tasks in sequential order or in parallel using ( Fork or Dynamic Fork). However, the next task in a pipeline can only be scheduled once the previous task was completed. We need to orchestrate a workflow where the task_1: is uploading a user file. The upload task can generate Streams of output. Each data record say corresponds to a page or view in the file (or say few frames in a video). We intend to schedule the next tasks, as soon as any data record is uploaded and without waiting for the whole file to be uploaded.

Is there a way to do using the Dynamic fork and other constructs in the conductor?
Is there a way to enhance conductor and support such use case? Maybe by adding a semi-completed state for the task? or by adding a new enhanced version of DynamicFork construct?

**Comment by kishorebanala** [comment]
@choubisat It is not possible with current Conductor task types. However, this can be achieved by adding new `StreamingFork` task, which works almost similar to Dynamic Fork, but unlike the later, it provides the input to spawned tasks through streams.

A `StreamingFork` could/should:

- Only spawn one task type (We can expand on this later, but better to start with limited implementation).
- Schedules new tasks as an update comes in.
- Is executed as a special form of Worker task (Unlike Dynamic Fork, which is a System task).
- Stays IN_PROGRESS after it schedules first task and until no more children are to be spawned.
- Has upper bounds on number of tasks it can spawn. For eg. A maximum or configurable default limit of 1000 tasks to being with.

If this is something that you think aligns with your needs, feel free to start a discussion regarding the implementation and we can continue from there.

**Comment by RAVIPMALL** [comment]
@kishorebanala  is it now possible? I want to process a csv in batches, a task that would fetch a batch and pass this batch to other tasks while the fetch batch continues for all the data in the csv.

**Comment by kishorebanala** [comment]
@RAVIPMALL This is not available, and is still a low priority for us. But, please feel free to open a discussion to gauge the interest and potentially a contribution from the community.
